proxy/backendtest/backendrecorder.go

```diff
@@ -60,6 +60,7 @@ func (rec *BackendRecorderHandler) GetURL() string {
 	return rec.server.URL
 }
 
-func (rec *BackendRecorderHandler) Close() {
+func (rec *BackendRecorderHandler) Done() {
```

Comment:
I think the func should not be renamed here, because Done() is used in stdlib waitgroup to signal that the goroutine has done the work. In this case we Close() our server if we get a signal via done channel.

Reply:
Before this change user of the Recorder would have to read (and block) on the done channel. This change wraps it into the method to also close the TestServer after wait time has elapsed. There is no need to have separate Close method.
I can rename `Done` to `Wait` if that feels better.

Reply:
Why `Done` to `Wait` and not to `Close`?

Reply:
> Why Done to Wait

Because users are supposed to call `recorder.Done()` at the point where they expect all the requests to complete in order to perform checks on the recorded data. If we call it `Close` there is a high chance that it would be used like
```go
recorder := NewBackendRecorder(10 * time.Millisecond)
defer recorder.Close()
```
which is wrong.